virtgpu_virgl.c

```diff
@@ -1157,6 +1157,10 @@ static uint32_t virgl_get_max_texture_2d_size(struct driver *drv)
 		return VIRGL_2D_MAX_TEXTURE_2D_SIZE;
 }
 
+static bool virgl_query_dev_feature() {
+       return (dev_feature == 2);
```

Comment:
It is easy to introduce problem by just checking the dev_feature counter to judge query_dev_feature enabling

Reply:
The feature check part:
			if (strcmp(params[i].name, "VIRTGPU_PARAM_QUERY_DEV") == 0) {
				dev_feature += params[i].value;
			}
			if (strcmp(params[i].name, "VIRTGPU_PARAM_RESOURCE_BLOB") == 0) {
				dev_feature += params[i].value;
			}
it doesn't collect feature before, i modify to check the driver name and blob feature.
if has blob feature and is virtio-gpu device, could use i915 to allocate buffer, otherwise need use virtio to allocate scanout buffer.
Do you have better solution?